components/LogWorkout/LogWorkout.jsx

```diff
@@ -76,6 +87,48 @@ const LogWorkout = (props) => {
 
   const [exerciseState, setExerciseState] = useState(initialExerciseState);
 
+  const currentUser = firebase.auth().currentUser.uid;
+  // const currentUser = '68w6wWz8l5QJO3tDukh1fRXWYjD2';
+
+  const dbRef = firebase.firestore();
+  const userRef = dbRef.collection('users').doc(currentUser);
+  const dispatch = useDispatch();
+
+  const incrementSelectedCycleIdx = () => {
+    userRef.update({ selectedCycleIndex: (cycleIdx + 1) % cycleLength });
+    dispatch({ type: INCREMENT_SELECTED_CYCLE_INDEX, cycleLength });
+  };
+
+  const sendWorkoutLogToDB = () => {
+    const workoutRecsRef = userRef.collection('workoutRecords');
+
+    const newWorkoutLog = {
+      workoutName: name,
+      workoutId,
+      date: `${format(new Date(), 'yyyy-MM-dd').toString()}`,
+      exercises: exerciseState.map((exercise) => ({
+        exerciseId: exercise.id,
+        exerciseName: exercise.name,
+        sets: exercise.sets.map((set) => {
+          const parsedSet = {
+            weight: (set.prevWeight === 'n/a' && set.weight === '') ? null : parseInt(set.weight) || parseInt(set.prevWeight),
+          };
+          _.set(parsedSet, exercise.type === 'REPS' ? ['reps'] : ['time'], parseInt(set.duration) || parseInt(set.prevDuration));
+          return parsedSet;
+        }),
+      })),
+    };
+
+    const newWorkoutDoc = {
+      lastPerformed: newWorkoutLog.date,
+      exercises: newWorkoutLog.exercises,
+    };
+
+    userRef.collection('workouts').doc(workoutId).update(newWorkoutDoc); // updates workout doc
+    dispatch(actions.workouts.updateWorkoutPrev(workoutId, newWorkoutLog.exercises)); // rerenders workout to show update prev details
+    workoutRecsRef.add(newWorkoutLog); // makes a new workoutRecord
+  };
```

Comment:
Is the code to put the new dates in Firebase in here?

Reply:
@shriyan44 @rohithdara not sure I didn't do this part I don't think

Reply:
I don't know what happened within the commits and pulls, but the call to the function that updated it was lost. I added it back in with the most recent commit.